site/content/api/Tenant-API.md

```diff
@@ -0,0 +1,294 @@
+| *tenant-id*              | *yes*     | *string*   |               | The ID of the tenant. |
+| *enabled*                | *no*      | *boolean*  | `true`        | If set to `false` the tenant is currently disabled. Protocol adapters MUST NOT allow devices of a disabled tenant to connect and MUST NOT accept data published by such devices.
+| *adapters*   | *yes*     | *array*    |               | A list of configuration options valid for certain adapters only. See [Adapter Configuration Format]({{< relref "#adapter-configuration-format" >}}) for details. **NB** This array must contain at least one element - an empty array is handled as an error. **NB** Only a single entry per *type* is allowed. If multiple entries for the same type are present it is handled as an error.
+
+If any of the mandatory members is either missing or present with invalid data in the payload object operations MUST return status code *`400` - Bad Request*.
+
+Additionally to the specified properties the JSON object MAY contain an arbitrary number of members with arbitrary names which can be of a scalar or a complex type. 
+This allows for future *well-known* additions and also allows *clients* to add further information which might be relevant to a *custom* adapter only.
+
+### Adapter Configuration Format
+
+The table below contains the properties which are used to configure a *Hono protocol adapter*:
+
+| Name                               | Mandatory | Type       | Default Value | Description |
+| :--------------------------------- | :-------: | :--------- | :------------ | :---------- |
+| *type*                       | *yes*     | *string*   | `null`        | The type of the adapter which this configuration belongs to |
+| *enabled*                          | *no*      | *string*   | `true`        | If set to false the tenant is not allowed to receive / send data utilizing the given adapter. |
+| *device-authentication-required*   | *no*      | *string*   | `true`        | If set to false, devices are not required to authenticate with the adapter before sending / receiving data. |
+
+Protocol Adapters SHOULD use the configuration properties set for a tenant when interacting with devices of that tenant, e.g. in order to make authorization decisions or to limit message rates per tenant etc.
+If for a certain protocol adapter no configuration is present it has the same semantic as a disabled protocol adapter with default values.
+
+Additionally to the specified properties the JSON object MAY contain an arbitrary number of members with arbitrary names which can be of a scalar or a complex type. 
+
+### Examples
+
+Below is an example for a payload defining an *enabled* tenant with id `DEFAULT_TENANT`.
+Devices belonging to the tenant can connect to Hono via the rest-adapter only and are required to authenticate with the adapter on connection.
+
+~~~json
+{
+  "tenant-id": "DEFAULT_TENANT",
+  "enabled": true,
+  "adapters": [
+    {
+      "type": "rest-adapter",
```

Comment:
Since it is already in the adapters array, the type "rest" (or "http" in the future) might be better?

Reply:
sound good to my ears. I renamed it :-)